test/extended/util/framework.go

```diff
@@ -0,0 +1,110 @@
+package util
+
+import (
+	"fmt"
+
+	kapi "github.com/GoogleCloudPlatform/kubernetes/pkg/api"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/fields"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/labels"
+	kutil "github.com/GoogleCloudPlatform/kubernetes/pkg/util"
+
+	buildapi "github.com/openshift/origin/pkg/build/api"
+	"github.com/openshift/origin/pkg/client"
+	"github.com/openshift/origin/pkg/util/namer"
+)
+
+// OsFramework provides helper functions and watchers for OpenShift
+type OsFramework struct {
+	Namespace *kapi.Namespace
```

Comment:
Why do we want to have one these for each namespace?  Also, what value does this bundling have for me?  Why wouldn't I simply have a `WaitForABuild` helper function that took a `BuildInterface` and a `buildName`?